config/crd/validation.yaml

```diff
@@ -0,0 +1,7 @@
+# PostgresCluster "v1beta1" is in "/spec/versions/0"
+
+# Containers should not run with a root GID.
+# - https://kubernetes.io/docs/concepts/security/pod-security-standards/
+- op: add
+  path: /spec/versions/0/schema/openAPIV3Schema/properties/spec/properties/supplementalGroups/items/minimum
+  value: 1
```

Comment:
:question: Is there any value to adding validation for the upper bound (i.e. 65535)?
The example PSP [here](https://kubernetes.io/docs/concepts/policy/pod-security-policy/#example-policies) seems to set a limit.

Reply:
I read a bit about this&sup1; while trying to find why we had 65534 hard-coded. Apparently, `uid_t` and `gid_t` used to be 16 bits and are now 32 bits. It looks like Kubernetes [chose `int64`](https://github.com/kubernetes/kubernetes/blob/v1.6.0/staging/src/k8s.io/apimachinery/pkg/types/unix_user_id.go) "as a safe bet against wrap-around" and [validates](https://github.com/kubernetes/kubernetes/blob/v1.22.0/pkg/apis/core/validation/validation.go#L3923-L3927) that values are `<= math.MaxInt32`.

🔧 I think we should add a maximum. No more than int32 (to match Kubernetes). 🤔 I see recommendations to stay within 16 bits, but I'm not certain we should enforce it. I'm inclined to let/expect users of this field Know What They Are Doing.

&sup1; I learned a lot from this page: https://systemd.io/UIDS-GIDS/

Reply:
I tried a value that was too large, and I think hit https://issue.k8s.io/97309. The StatefulSet was accepted/valid, but it could not create the Pod:

```
Pod "hippo-00-z9fc-0" is invalid: spec.securityContext.supplementalGroups[0]: Invalid value: 4000000000: must be between 0 and 2147483647, inclusive
```

Reply:
Added a maximum with links. [`bc27ad65`](https://github.com/CrunchyData/postgres-operator/pull/2593/commits/bc27ad6524d55c4a8f3c993c1871064be9056a96)